connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConfigBackingStore.java

```diff
@@ -51,8 +52,9 @@ public interface ConfigBackingStore {
      * Update the configuration for a connector.
      * @param connector name of the connector
      * @param properties the connector configuration
+     * @param callback the callback to be invoked after the put is complete; can be {@code null} if no callback is desired
      */
-    void putConnectorConfig(String connector, Map<String, String> properties);
+    void putConnectorConfig(String connector, Map<String, String> properties, Callback<Void> callback);
```

Comment:
Could potentially add a new overloaded method in `KafkaConfigBackingStore` and avoid touching this interface method although that would require refactoring `AbstractHerder` to make it generic in order to allow `DistributedHerder` to access an instance of `KafkaConfigBackingStore` rather than `ConfigBackingStore`. However, that would be a much more noisy change IMO.